python/python/glide/async_commands/server_modules/json.py

```diff
@@ -84,6 +84,33 @@ def to_args(self) -> List[str]:
         return args
 
 
+class JsonArrPopOptions:
+    """
+    Options for the JSON.ARRPOP command.
+
+    Args:
+        path (TEncodable): The path within the JSON document.
+        index (Optional[int]): The index of the element to pop. If not specified, will pop the last element.
+            Out of boundary indexes are rounded to their respective array boundaries. Defaults to None.
+    """
+
+    def __init__(self, path: TEncodable, index: Optional[int] = None):
```

Comment:
I get confused each time i review json module commands? Is path optional or required? because for this commend it is marked as optional

Reply:
path is optional but if you specify index you must specify path, so I have put them together inside an options class, where you must specify path (if you choose to specify options) but you dont have to specify index

commands syntax:
`JSON.ARRPOP <key> [path [index]]`